### PR TITLE
Fixes datetime issue with untyped object.

### LINKED ISSUE #1060: DateTimeValue(UntypedObject) does not honor time zone
Thanks for addressing date time time zones in recent code check-in! While attempting to consume the latest NuGet, I found the below issue:

`DateTimeValue(""2022-11-21T12:13:30Z"")` honors time zone as expected, but
`DateTimeValue(ToUntypedObject(""2022-11-21T12:13:30Z""))` does not. (`ToUntypedObject` is a custom function defined below)

Repro steps:
```csharp

        [Fact]
        public async Task DateTimeTests()
        {
            var powerFxConfig = new PowerFxConfig(System.Globalization.CultureInfo.InvariantCulture);
            powerFxConfig.AddFunction(new ToUntypedObjectFunction());

            var engine = new RecalcEngine(powerFxConfig);

            var runtimeConfig = new RuntimeConfig();
            runtimeConfig.SetTimeZone(System.TimeZoneInfo.Utc);

            // DateTimeValue(String) works as expected
            var result = await engine.EvalAsync(@"DateTimeValue(""2022-11-21T12:13:30Z"")", default, null, null, runtimeConfig).ConfigureAwait(false);
            var dateTimeValue = Assert.IsType<DateTimeValue>(result);
            var dateTimeValueConverted = dateTimeValue.GetConvertedValue(System.TimeZoneInfo.Utc);
            Assert.Equal("2022-11-21T12:13:30.0000000Z", dateTimeValueConverted.ToString("o")); // this succeeds

            // DateTimeValue(UntypedObject) doesn't work
            result = await engine.EvalAsync(@"DateTimeValue(ToUntypedObject(""2022-11-21T12:13:30Z""))", default, null, null, runtimeConfig).ConfigureAwait(false);
            dateTimeValue = Assert.IsType<DateTimeValue>(result);
            dateTimeValueConverted = dateTimeValue.GetConvertedValue(System.TimeZoneInfo.Utc);
            Assert.Equal("2022-11-21T12:13:30.0000000Z", dateTimeValueConverted.ToString("o")); // this fails
        }

        private class ToUntypedObjectFunction : ReflectionFunction
        {
            public ToUntypedObjectFunction()
                : base("ToUntypedObject", FormulaType.UntypedObject, FormulaType.String)
            {
            }

            public FormulaValue Execute(StringValue val)
            {
                return FormulaValue.New(new SimpleObject(val));
            }
        }

        private class SimpleObject : IUntypedObject
        {
            private readonly FormulaValue _value;

            public SimpleObject(FormulaValue value)
            {
                _value = value;
            }

            public IUntypedObject this[int index] => throw new NotImplementedException();

            public FormulaType Type => _value.Type;

            public int GetArrayLength()
            {
                throw new NotImplementedException();
            }

            public bool GetBoolean()
            {
                return ((BooleanValue)_value).Value;
            }

            public double GetDouble()
            {
                return ((NumberValue)_value).Value;
            }

            public string GetString()
            {
                return ((StringValue)_value).Value;
            }

            public bool TryGetProperty(string value, out IUntypedObject result)
            {
                throw new NotImplementedException();
            }
        }
```

I think the behavior can be fixed by setting the `DateTimeStyles` to `RoundTripKind` instead of `None`. Similarly for DateValue() and TimeValue().

https://github.com/microsoft/Power-Fx/blob/caf096360b425a04b53373b8e32678cb8fe508e0/src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs#L241

### PR DESCRIPTION
**Pull request by jas-valgotar**
Fixes #1060

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/DateTimeUTCTests.cs [hunk 1]
```diff
@@ -160,5 +161,165 @@ public void GetConvertedDateTimeValueTest()
...
         }
+
+        [Fact]
+        public async Task DateTimeValueTests()
+        {
+            _engine.Config.AddFunction(new ToUntypedObjectFunction());
+
+            var expectedUTC = new DateTime(2022, 11, 21, 12, 13, 30, DateTimeKind.Utc);
+            
+            //var expectedLocal = new DateTime(2022, 11, 21, 12, 13, 30, DateTimeKind.Local);
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/DateTimeUTCTests.cs:172]: >var expectedLocal = new DateTime(2022, 11, 21, 12, 13, 30, DateTimeKind.Local); [](http://example.com/codeflow?start=14&length=79)

Remove commented out code

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/DateTimeUTCTests.cs [hunk 2]
```diff
@@ -160,5 +161,165 @@ public void GetConvertedDateTimeValueTest()
...
+        }
+
+        [Fact]
+        public async Task DateValueTests()
+        {
+            _engine.Config.AddFunction(new ToUntypedObjectFunction());
+
+            var expectedUTC = new DateTime(2022, 11, 21, 22, 13, 30, DateTimeKind.Utc);
+
+            //var expectedLocal = new DateTime(2022, 11, 21, 12, 13, 30, DateTimeKind.Local);
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/DateTimeUTCTests.cs:225]: >//var expectedLocal = new DateTime(2022, 11, 21, 12, 13, 30, DateTimeKind.Local); [](http://example.com/codeflow?start=12&length=81)

Remove commented out code

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**